src/training/validator.h

```diff
@@ -58,31 +57,33 @@ class ValidatorBase : public TrainingObserver {
 template <class DataSet>
 class Validator : public ValidatorBase {
 public:
-  Validator(std::vector<Ptr<Vocab>> vocabs,
-            Ptr<Config> options,
-            bool lowerIsBetter = true)
-      : ValidatorBase(lowerIsBetter), vocabs_(vocabs), options_(options) {}
+  Validator(std::vector<Ptr<Vocab>> vocabs, Ptr<Options> options, bool lowerIsBetter = true)
+      : ValidatorBase(lowerIsBetter),
+        vocabs_(vocabs),
+        // options_ is a copy of global options, so it can be safely modified within the class
+        options_(New<Options>(*options)) {
+    // set options common for all validators
```

Comment:
This is fixing the issue is asked about above? Is this correctly being propagated?
We should indeed have a clone() function.

Reply:
Yes, it should enable subclasses to conveniently modify options_. It's seems to be correctly propagated as tests fail without this.